src/datastore/pg-write-store.ts

```diff
@@ -1700,17 +1714,47 @@ export class PgWriteStore extends PgStore {
     return txIds;
   }
 
+  private _debounceMempoolStat: {
+    lastTrigger?: number | null;
+    debounce?: NodeJS.Timeout | null;
+    running: boolean;
+  } = { running: false };
+  /**
+   * Debounce the mempool stat process in case new transactions pour in.
+   */
+  private debounceMempoolStat() {
+    this._debounceMempoolStat.lastTrigger = Date.now();
+    if (this._debounceMempoolStat.running) return;
+    if (this._debounceMempoolStat.debounce != null) {
+      clearTimeout(this._debounceMempoolStat.debounce);
+      this._debounceMempoolStat.debounce = null;
+    }
+    this._debounceMempoolStat.debounce = setTimeout(async () => {
+      this._debounceMempoolStat.running = true;
+      this._debounceMempoolStat.lastTrigger = null;
+      try {
+        const mempoolStats = await this.getMempoolStatsInternal({ sql: this.sql });
+        this.eventEmitter.emit('mempoolStatsUpdate', mempoolStats);
+      } catch (e) {
+        logger.error(e, `failed to reconcile mempool`);
+      } finally {
+        this._debounceMempoolStat.running = false;
+        if (this._debounceMempoolStat.lastTrigger != null) {
+          this.debounceMempoolStat();
+        }
+      }
+    }, 1000);
```

Comment:
This can default to 1000, but can you make it a configurable env var?

Reply:
Done.

Reply:
Please also add associated entry to `.env` file

Reply:
Added, please review the wording.

Reply:
LGTM thanks!